### PR TITLE
Fix Aff_transformation_3 missing includes

### LINKED ISSUE #4382: Build errors with CGAL/Aff_transformation_3.h in CGAL 4.14.2
## Issue Details

Build errors with CGAL 4.14.2 when including `CGAL/Aff_transformation_3.h`.

## Source Code

```c
#include <CGAL/Aff_transformation_3.h>
```

compiling with GCC 9.2.0:
```
In file included from test.cpp:1:
/usr/include/CGAL/Aff_transformation_3.h:86:6: error: ‘ostream’ in namespace ‘std’ does not name a type
   86 | std::ostream&
      |      ^~~~~~~
/usr/include/CGAL/Aff_transformation_3.h:30:1: note: ‘std::ostream’ is defined in header ‘<ostream>’; did you forget to ‘#include <ostream>’?
   29 | #include <CGAL/aff_transformation_tags.h>
  +++ |+#include <ostream>
   30 | 
/usr/include/CGAL/Aff_transformation_3.h:96:6: error: ‘istream’ in namespace ‘std’ does not name a type
   96 | std::istream&
      |      ^~~~~~~
/usr/include/CGAL/Aff_transformation_3.h:30:1: note: ‘std::istream’ is defined in header ‘<istream>’; did you forget to ‘#include <istream>’?
   29 | #include <CGAL/aff_transformation_tags.h>
  +++ |+#include <istream>
   30 |
```

## Environment

* Operating system: Arch Linux
* Compiler: GCC 9.2.0
* Release or debug mode: probably Release mode, using the prebuilt package https://www.archlinux.org/packages/community/x86_64/cgal/
* Specific flags used (if any): none
* CGAL version: 4.14.2
* Boost version: 1.71.0 (https://www.archlinux.org/packages/extra/x86_64/boost/)
* Other libraries versions if used (Eigen, TBB, etc.): none

## Further notes

The reason seems to be commit 856b7722f1038e333eb9a25e9f87a7b986d8f251 which fixes a typo in the macro name `BOOST_VERSION`. As a result, the header file `boost/container/flat_map.hpp` is not included anymore and doesn't pull in `std::istream` nor `std::ostream` anymore.
The correct solution is probably to include the headers mentioned by the compiler, or maybe `iosfwd` is enough.

### PR DESCRIPTION
**Pull request by maxGimeno**
## Summary of Changes
Add misisng headers in Aff_transformation_3.h
## Release Management

* Affected package(s):Kernel_23
* Issue(s) solved (if any): fix #4382

### CODE REVIEW

File: Surface_mesh_simplification/include/CGAL/Surface_mesh_simplification/Detail/Edge_collapse.h [hunk 1]
```diff
@@ -136,7 +136,7 @@ class EdgeCollapse
   } ;
   
   
-  typedef Modifiable_priority_queue<halfedge_descriptor,Compare_cost,edge_id> PQ ;
+  typedef Modifiable_priority_queue<halfedge_descriptor,Compare_cost,edge_id> PQ;
   typedef typename PQ::handle pq_handle ;
```
Comment (MaelRL) [inline Surface_mesh_simplification/include/CGAL/Surface_mesh_simplification/Detail/Edge_collapse.h]: alien changes in the whole file

Comment (maxGimeno) [reply Surface_mesh_simplification/include/CGAL/Surface_mesh_simplification/Detail/Edge_collapse.h]: My IDE removes trailing white spaces when I save :(

Comment (maxGimeno) [reply Surface_mesh_simplification/include/CGAL/Surface_mesh_simplification/Detail/Edge_collapse.h]: Wait a sec, why was this file modified at all ? oO

Comment (MaelRL) [reply Surface_mesh_simplification/include/CGAL/Surface_mesh_simplification/Detail/Edge_collapse.h]: You can use `git add -p` to have better control of what goes into the commit. It's pretty fast once you're used to the main commands (y/n/d/a, mainly) and you can also split changes from a single file into multiple commits.

**Comment by maxGimeno** [comment]
Successfully tested in https://cgal.geometryfactory.com/CGAL/testsuite/results-5.1-Ic-35.shtml

**Pull request merged**